wadsrc/static/zscript/actors/actor.zs

```diff
@@ -534,7 +534,9 @@ class Actor : Thinker native
 	// [AA] Called by inventory items at the end of CallTryPickup to let actors
 	// do something with the items they've received. 'Item' might be null for
 	// items that disappear on pickup.
-	virtual void HasReceived(Inventory item) {}
+	// 'itemcls' is passed unconditionally, so it can still be read even if
+	// 'item' is null due to being destroyed with GoAwayAndDie() on pickup.
+	virtual void HasReceived(Inventory item, class<Inventory> itemcls) {}
```

Comment:
You will need to make the parameters optional for this to work. To do this, change the code from
`	virtual void HasReceived(Inventory item, class<Inventory> itemcls) {}`
to
`	virtual void HasReceived(Inventory item = null, class<Inventory> itemcls = null) {}`

Otherwise, it will break backwards compatibility.

Reply:
To clarify, only `itemcls` needs to be null. The first parameter was never optional so everyone had to specify it anyway when defining the method signature.

Reply:
Good catch, thanks!